include/fmt/prepare.h

```diff
@@ -246,14 +246,19 @@ class prepared_format {
   }
 
  private:
-  typedef typename buffer_context<char_type>::type context;
+  template <typename Range, typename Context, typename... StoreArgs>
+  auto vformat_to(Range out, format_arg_store<Context, StoreArgs...> args) const
```

Comment:
`vformat*` functions should take `basic_format_args` not `format_arg_store`.

Reply:
Those are private methods. I've introduced the overload that takes `format_arg_store`, because calling `prepared_format::format_to()` with `char*` as an output iterator, introduced compilation error `could not match 'basic_format_args' against 'format_arg_store'`. That's why I decided to create the overload, explicitly create `basic_format_args` there and pass it to the proper `vformat` implementation.

Does it resolve your doubts or it's still wrong? If it's wrong, what do you propose?

Reply:
The matching error is because template argument deduction doesn't play well with implicit conversion from `format_arg_store` to `format_args`. I suggest removing the extra overload and adding braces to force the conversion as it is done in https://github.com/fmtlib/fmt/blob/87fbc6f7566e4d3266bd3a2cd69f6c90e1aefa5d/include/fmt/core.h#L1357

Reply:
Hmm, I think that's not exactly the case. Braces helps here because with such call, compiler doesn't need to deduce `context` for `basic_format_args`, thus whole `basic_format_args<context>` type is known. It's known because `Char` type known, as well.
https://github.com/fmtlib/fmt/blob/87fbc6f7566e4d3266bd3a2cd69f6c90e1aefa5d/include/fmt/core.h#L1317

In my case, the `context` type is not known and needs to be deduced, so `basic_format_args` is not known and can not be deduced because compiler would need to perform implicit conversion.

To handle this, I've added explicit call to `basic_format_args` ctors. Please check it and let me know if it looks reasonable.